Call ledger version of `evaluateTxBalance` from `balanceTx`.

jonathanknowles opened the pull request:
## Issue Number

https://input-output.atlassian.net/browse/ADP-2651

## Summary

This PR:
- adjusts `balanceTx` (and inner functions) to call the `cardano-ledger` version of the `evaluateTxBalance` function instead of the `cardano-api` version.
- removes `evaluateTxBalance` from `TxLayer`.

jonathanknowles committed:
Add function `fromCardanoTx` to `Wallet.Write.Tx`.

jonathanknowles committed:
Add function `evaluateTransactionBalance` to `Write.Tx`.

jonathanknowles committed:
Call `Write.Tx.evaluateTransactionBalance` from `balanceTransaction`.

jonathanknowles committed:
Add function `cardanoValueFromCoreValue` to `Write.Tx`.

jonathanknowles committed:
Remove `evaluateTransactionBalance` from `TransactionLayer`.

Anviking commented on lib/wallet/src/Cardano/Wallet/Write/Tx.hs:
```diff
@@ -752,6 +771,46 @@ fromCardanoUTxO = withStandardCryptoConstraint (recentEra @era) $
   where
     unCardanoUTxO (Cardano.UTxO m) = m
 
+cardanoValueFromCoreValue
```
```suggestion
toCardanoValue
```

jonathanknowles replied:
Fixed in 32a1fb80fc8aac4758720bdcebdc733ae7904b21.

jonathanknowles committed:
Rename `cardanoValueFromCoreValue` to `toCardanoValue`.

In response to review feedback:

https://github.com/input-output-hk/cardano-wallet/pull/3752#pullrequestreview-1305248238

Anviking commented on lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs:
```diff
@@ -159,6 +159,7 @@ import qualified Cardano.Wallet.Primitive.Types.UTxO as W
 import qualified Cardano.Wallet.Primitive.Types.UTxO as UTxO
 import qualified Cardano.Wallet.Primitive.Types.UTxOIndex as UTxOIndex
 import qualified Cardano.Wallet.Primitive.Types.UTxOSelection as UTxOSelection
+import qualified Cardano.Wallet.Write.Tx as Write.Tx
```
This import already exists as `WriteTx`
```suggestion
```

Since Yura also seemed to default to this, I added this topic to next Monday's meeting, but could we stick to `WriteTx` for now?

jonathanknowles replied:
Added to the next meeting minutes!

Anviking commented on lib/wallet/src/Cardano/Wallet/Write/Tx.hs:
```diff
@@ -116,9 +118,13 @@ module Cardano.Wallet.Write.Tx
+    , toCardanoValue
 
+    -- * Balancing
+    , evaluateTransactionBalance
```
I suppose it could have gone into `Write.Tx.Balance` too 🤔

jonathanknowles replied:
> I suppose it could have gone into `Write.Tx.Balance` too thinking

Definitely! We might need to revise our function and module structure at a certain point.

Anviking reviewed (approved):
👍 Awesome, thanks! Just could we fix this https://github.com/input-output-hk/cardano-wallet/pull/3752/files#r1111714093 ?
